lib/app/services/media_service/media_compress_service.dart

```diff
@@ -0,0 +1,202 @@
+  }
+
+  Future<void> _initWindowsPackage() async {
+    final success = await FFMpegHelper.instance.setupFFMpegOnWindows();
+    if (!success) {
+      Logger.log('Failed to setup ffmpeg on Windows');
+      throw Exception('Failed to setup ffmpeg on Windows');
+    }
+  }
+
+  Future<XFile> _compressVideoForWeb(XFile inputFile, List<CustomArgument> args) async {
+    const input = 'input.mp4';
+    const output = 'output.mp4';
+    final ffmpeg = createFFmpeg(CreateFFmpegParam());
+
+    if (!ffmpeg.isLoaded()) {
+      await ffmpeg.load();
+    }
+
+    ffmpeg.writeFile(input, await inputFile.readAsBytes());
+
+    await ffmpeg.runCommand(
+      "${FFMpegInput.asset(input).toCli()} ${args.map((arg) => FFMpegInput(arg.toArgs()).toCli()).join(' ')} $output",
+    );
+
+    final compressedVideoBytes = ffmpeg.readFile(output);
+    return XFile.fromData(compressedVideoBytes, name: output);
+  }
+
+  Future<XFile> compressImage(
+    XFile file, {
+    required int quality,
+    required Size size,
+    bool keepAspectRatio = true,
+  }) async {
+    try {
+      final resizedBytes = await compute<ResizeImageParams, Uint8List>(
+        (params) async {
+          return resizeImage(params.file, params.size, keepAspectRatio: params.keepAspectRatio);
+        },
+        ResizeImageParams(file: file, size: size, keepAspectRatio: keepAspectRatio),
+      );
+
+      if (resizedBytes.isEmpty) {
+        throw Exception('Failed to resize image.');
+      }
+
+      final config = compressor_package.ImageFileConfiguration(
+        input: compressor_package.ImageFile(
+          filePath: '', // File path left empty since we're using rawBytes
+          rawBytes: resizedBytes,
+        ),
+        config: compressor_package.Configuration(
+          quality: quality,
+        ),
+      );
+
+      final output = await compressor_package.compressor.compress(config);
+
+      return XFile.fromData(output.rawBytes, name: file.name, mimeType: file.mimeType);
+    } catch (e) {
+      Logger.log('Error during image compression: $e');
+      rethrow;
+    }
+  }
+
+  Future<Uint8List> resizeImage(XFile file, Size size, {bool keepAspectRatio = true}) async {
+    final originalBytes = await file.readAsBytes();
+    final decodedImage = img.decodeImage(originalBytes);
+
+    if (decodedImage == null) {
+      throw Exception('Failed to decode image.');
+    }
+
+    final resizedImage = img.copyResize(
+      decodedImage,
+      width: size.width.toInt(),
+      height: size.height.toInt(),
+      maintainAspect: keepAspectRatio,
+    );
+
+    return Uint8List.fromList(img.encodePng(resizedImage));
+  }
+
+  Future<String> _generateOutputPath({bool isThumbnail = false}) async {
+    // Get a platform-independent temporary directory
+    final tempDir = await getApplicationCacheDirectory();
+
+    // Generate a new output filename
+    final timestamp = DateTime.now().millisecondsSinceEpoch;
+    final extension = isThumbnail ? 'jpg' : 'mp4';
```

Comment:
Good catch! Since the thumbnail creator FFmpeg function only works with JPG, I've converted it to WebP during creation, so now we'll get a WebP at the end of our function.